arrow-csv/src/writer.rs

```diff
@@ -88,6 +88,25 @@ where
     lexical_to_string(c.value(i))
 }
 
+fn invalid_cast_error(dt: String, col_index: usize, row_index: usize) -> ArrowError {
```

Comment:
```suggestion
fn invalid_cast_error(dt: &str, col_index: usize, row_index: usize) -> ArrowError {
```
Might simplify some things